### PR TITLE
Experiment - Performance Transaction set Manually

### PR DESCRIPTION
**Pull request by thinkocapo**
**Do Not Merge** this

for experimentation and feedback purposes only.

sdk.version `4.0.0-alpha.2`
sdk.name `sentry.java.android`

add lots of lifecycles/callbacks, ideas for spans:
https://developer.android.com/topic/libraries/architecture/lifecycle
child for onCreate onStart onDestroy
finish transaction on onPause onStop

activity lifecycles, view lifecycles, inflation, onClick callbacks.

## Based on
https://github.com/getsentry/sentry-java/pull/1058

## Testing
[MainActivity.onCreate Transaction](https://sentry.io/organizations/testorg-az/discover/paranoid-android:bda1287c25c84017aad6cbef7d85d2cd/?field=title&field=event.type&field=project&field=user.display&field=timestamp&name=All+Events&project=261820&project=1801383&query=event.type%3Atransaction&sort=-timestamp&statsPeriod=14d&widths=-1&widths=-1&widths=-1&widths=-1&widths=-1) (no inner spans set yet)

### CODE REVIEW

**Review by ndmanvar — APPROVED** [review]

File: app/build.gradle [hunk 1]
```diff
@@ -52,7 +52,7 @@ dependencies {
     implementation fileTree(dir: 'libs', include: ['*.jar'])
     implementation 'androidx.appcompat:appcompat:1.1.0'
     implementation group: 'androidx.constraintlayout', name: 'constraintlayout', version: '1.1.3'
-    implementation 'io.sentry:sentry-android:3.1.0'
+    implementation 'io.sentry:sentry-android:4.0.0-alpha.2'
```
Comment (marandaneto) [inline app/build.gradle:55]:
```suggestion
    implementation 'io.sentry:sentry-android:4.0.0'
```

File: app/src/main/java/com/example/vu/android/MyApplication.java [hunk 2]
```diff
@@ -41,7 +39,14 @@ public void onCreate() {
...
                     return event;
             });
+            options.setTracesSampleRate(1.0);
         });
+
+//        SentryTransaction activityTransaction = Sentry.startTransaction("MainActivity.onCreate");
+//        activityTransaction.finish();
+
+        // we now enable this in AndroidManifest.xml
+        // options.setEnableSessionTracking(true);
```
Comment (marandaneto) [inline app/src/main/java/com/example/vu/android/MyApplication.java:49]: this is enabled by default, could be removed

File: app/src/main/java/com/example/vu/android/MyApplication.java [hunk 3]
```diff
@@ -41,7 +39,14 @@ public void onCreate() {
                 else
                     return event;
             });
+            options.setTracesSampleRate(1.0);
```
Comment (marandaneto) [inline app/src/main/java/com/example/vu/android/MyApplication.java:42]: already done in the `AndroidManifest.xml`, no need to duplicate

File: app/src/main/java/com/example/vu/android/MyApplication.java [hunk 4]
```diff
@@ -41,7 +39,14 @@ public void onCreate() {
...
         });
+
+//        SentryTransaction activityTransaction = Sentry.startTransaction("MainActivity.onCreate");
+//        activityTransaction.finish();
```
Comment (marandaneto) [inline app/src/main/java/com/example/vu/android/MyApplication.java:46]: better done in the `MainActivity.java `

**Pull request closed without merging**